src/collectives/device/msccl_interpreter.h

```diff
@@ -68,7 +68,6 @@ namespace {
     // User pointers for primitives
     T* thisInput = (T*)args->sendbuff;
     T* thisOutput = (T*)args->recvbuff;
-    T* thisScratch = (T*)ncclShmem.mscclShmem.scratchBuffer;
```

Comment:
not sure why you removed this. is this beneficial to the performance?

Reply:
It is not, just removed it because if the primitive does not have its input or output buffer being the scratch one, this will end up being a dead code, that is the reason I have allowed it only when needed. The others are used along the code. I can return it to the code if you prefer.

Reply:
ah OK, yeah makes sense. However, accessing ncclShmem.mscclShmem.scratchBuffer inside the loop in line 127/128 might be more costly that using a cache value. So, that's why I am asking to revert this change :)

Reply:
right! Done!